torchmultimodal/models/gpt.py

```diff
@@ -123,6 +137,9 @@ def forward(
             x_out = self.out_token_emb(out_modality) + self.out_pos_emb(out_pos_ids)
             x = torch.cat((x_in, x_out), dim=1)
 
+        if self.training or right_shift:
```

Comment:
Noob question: when would someone want right_shift turned on if not in training mode?

Reply:
Since we right shift during training mode, we will need to right shift the first token during generation to be consistent. For the latter because we are only interested in the first token, we will need to tell the generation utility (which takes the GPT model as input) to trigger it at `decode_step=0` and switch it off for the subsequent steps.

Context:
- MUGEN handles this by piping in `decode_step` throughout their transformer stack which not only makes the APIs look clunky but also introduces unnecessary branching codepaths inside, for example, the original `RightShift` forward pass was merely a passthrough when `decode_step>0` [here](https://github.com/mugen-org/MUGEN_baseline/blob/b45da493f3ddbcced52d44eee02d443f90842474/lib/models/gpt/attention.py#L318).